frontend/catalyst/debug/compiler_functions.py

```diff
@@ -160,3 +160,35 @@ def compile_from_mlir(ir, compiler=None, compile_options=None):
         result_types = [mlir.ir.RankedTensorType.parse(rt) for rt in func_data[1].split(",")]
 
     return CompiledFunction(shared_object, qfunc_name, result_types, None, compiler.options)
+
+
+@debug_logger
+def get_pipeline_output(fn, pass_name):
+    """Capture IR string from the given compiler pass.
+
+    Args:
+        fn (QJIT): a qjit-decorated function
+        pass_name (str): target compiler pass name
+
+    Returns:
+        str: output ir from the target compiler pass
+    """
+    if not fn.compiler.options.keep_intermediate:
+        raise RuntimeError("keep_intermediate must be set to True to get pipeline's output.")
+    if pass_name == "last":
+        return fn.compiler.last_compiler_output.get_output_ir()
+    return fn.compiler.get_output_of(pass_name)
+
+
+@debug_logger
+def replace_ir(fn, pass_name, new_ir):
+    """Specify new IR that will be used for future compilation.
```

Comment:
Since this one also accepts stages it should include the snippet you added above I think :) 
```suggestion
    """Replace the IR at any compilation stage that will be used the next time the function runs.

    It is important that the function signature (inputs & outputs) for the next execution matches
    that of the provided IR, or else the behaviour is undefined.

    All the available stages are:
        - MILR: mlir, HLOLoweringPass, QuantumCompilationPass, BufferizationPass, and
                MLIRToLLVMDialect.
        - LLVM: llvm_ir, CoroOpt, O2Opt, Enzyme, and last.
    Note that `CoroOpt` (Coroutine lowering), `O2Opt` (O2 optimization), and `Enzyme` (Automatic
    differentiation) passes do not always happen. `last` denotes the stage right before object file
    generation.
```